ewatercycle/models/wflow.py

```diff
@@ -0,0 +1,157 @@
+        shutil.copytree(src=cfg_dir, dst=work_dir)
+        self.cfg_dir = work_dir.resolve()
+        print(f"Working directory created: {work_dir}")
+
+    def _setup_cfg_file(self, cfg_file: PathLike, **kwargs):
+        cfg = CaseConfigParser()
+        cfg.read(cfg_file)
+        self.cfg = cfg
+
+        for section, options in kwargs.items():
+            for option, value in options.items():
+                cfg.set(section, option, value)
+
+        new_cfg_file = Path(self.cfg_dir) / "wflow_ewatercycle.ini"
+        with new_cfg_file.open("w") as filename:
+            cfg.write(filename)
+
+        self.cfg_file = new_cfg_file.resolve()
+        print(f"Created {self.cfg_file}.")
+
+    def _setup_forcing(self, forcing_data: Optional[PathLike]):
+        if forcing_data is not None:
+            shutil.copy(src=forcing_data, dst=self.cfg_dir)
+            self.cfg.set("framework", "netcdfinput", Path(forcing_data).name)
+
+            with self.cfg_file.open("w") as filename:
+                self.cfg.write(filename)
+            print(
+                f"Copied forcing data to {self.cfg_dir} and updated {self.cfg_file} accordingly."
+            )
+
+    def _start_container(self):
+        if CFG["container_engine"] == "docker":
+            self.bmi = BmiClientDocker(
+                image=CFG["docker_images.wflow"],
+                image_port=55555,
+                work_dir=str(self.cfg_dir),
+                timeout=10,
+            )
+        elif CFG["container_engine"] == "singularity":
+            image = CFG["singularity_images.wflow"]
+
+            message = f"No singularity image found at {image}"
+            assert Path(image).exists(), message
+
+            self.bmi = BmiClientSingularity(
+                image=image,
+                work_dir=str(self.cfg_dir),
+                timeout=10,
+            )
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+        print(f"Started wflow container with working directory {self.cfg_dir}")
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_y(grid),
+                "latitude": self.bmi.get_grid_x(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da.where(da != -999)
```

Comment:
Can get_value be overwritten here to also replace -999 with na?

```suggestion
        return da.where(da != -999)

    def get_value(name: str) -> np.ndarray:
         da = super(name).get_value()
         return da.where(da != -999)
```

Reply:
Hmm I don't really like that. `get_value` directly interfaces the model fiels, whereas `get_value_as_xarray` is a convenience function that helps the user by tranforming it to something more user friendly.

I changed the inline comments, hope this is clearer. The ideal solution of course is to fix it in wflow.

Reply:
Yep, comment in notebook clarifies it is the fault of the model not the ewatercycle or grpc4bmi package.